### PR TITLE
Bring sanity to list_associated_files

### PR DESCRIPTION
**Pull request by medariox**
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read [contribution guide](https://github.com/PyMedusa/SickRage/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: medusa/post_processor.py [hunk 1]
```diff
@@ -209,14 +198,14 @@ def recursive_glob(treeroot, pattern):
                     new_file_name = file_name
                     sub_file_name = file_name.rpartition('.')[0]
 
-                if is_subtitle and sub_file_name.lower() == base_name.lower().replace('[[]', '[').replace('[]]', ']'):
+                if is_subtitle and sub_file_name.lower() == base_name.lower():
```
Comment (fernandog) [inline medusa/post_processor.py]: @medariox  need to find why this was added (blame) and if we can safely remove now

Comment (medariox) [inline medusa/post_processor.py]: It's only if glob is used and even then it's not needed in this case.

File: medusa/post_processor.py [hunk 2]
```diff
@@ -17,20 +17,19 @@
...
 
 import adba
 
-from babelfish import language_converters
-
 import medusa as app
 
 from six import text_type
 
+from subtitles import from_code, from_ietf_code
```
Comment (ratoaq2) [inline medusa/post_processor.py]: It should be
`from .subtitles`

File: medusa/post_processor.py [hunk 3]
```diff
@@ -148,85 +147,54 @@ def list_associated_files(self, file_path, base_name_only=False,
...
 
-            # loop through all the files in the folder, and check if they are the same name
-            # even when the cases don't match
-            for filefound in checklist:
+            file_name = os.path.basename(found_file).lower()
 
-                file_name = filefound.rpartition('.')[0]
-                file_extension = filefound.rpartition('.')[2]
-                is_subtitle = None
+            if file_name[-3:] in subtitle_extensions:
```
Comment (ratoaq2) [inline medusa/post_processor.py]: Shouldn't it be everything after the last dot?

File: medusa/post_processor.py [hunk 4]
```diff
@@ -148,85 +147,54 @@ def list_associated_files(self, file_path, base_name_only=False,
...
         for associated_file_path in filelist:
             # Exclude the video file we are post-processing
             if associated_file_path == file_path:
                 continue
 
-            # Exlude non-subtitle files with the 'only subtitles' option (not implemented yet)
-            # if subtitles_only and not associated_file_path[-3:] in subtitle_extensions:
-            #     continue
+            # Exlude non-subtitle files with the 'only subtitles' option
+            if subtitles_only and not associated_file_path[-3:] in subtitle_extensions:
```
Comment (ratoaq2) [inline medusa/post_processor.py]: Same as before. Probably we should have a method: is_subtitle, like the one: isMediaFile

**Comment by medariox** [comment]
Thanks for the review @ratoaq2.
I agree that [-3:] is not a reliable way to get the subtitle extension, it should always be .rpartition('.')[2] instead. Was thinking about implementing is_subtitle, as we have also other atrocities further down the file, that could surely benefit from it.

There are a few questions that I'd like to discuss before adding the method:
1) What should be supported? E.g. should just passing the extension 'srt' or even '.srt' be able to determine if it's a subtitle as well or just file_name and full_path?
2) What about language subs? Should we differentiate about the two types (maybe optionally) or should we just check the extension?

I'd like some feedback about this before continuing.

**Comment by ratoaq2** [comment]
Imo we should have like a fileutils that works on filepaths:
- get_extension(filepath): return 'abc' 
- is_subtitle(filepath) : return True
- get_subtitle_language(filepath): return Language('en')

**Comment by medariox** [comment]
@ratoaq2 
Is this more or less what you were thinking of? https://github.com/pymedusa/Medusa/pull/1509/commits/9022783283278a11c5b663e0e2381cbb3ba8034b

**Comment by ratoaq2** [comment]
Exactly

**Pull request merged**